lib/minitest/reporters/mean_time_reporter.rb

```diff
@@ -0,0 +1,201 @@
+      end
+
+      protected
+
+      attr_accessor :all_suite_times
+
+      private
+
+      # @return [Hash<String => Float>]
+      def current_run
+        Hash[all_suite_times]
+      end
+
+      # @return [Hash] Sets default values for the filenames used by this class.
+      def defaults
+        {
+          previous_runs_filename: '/tmp/minitest_reporters_previous_run',
+          report_filename:        '/tmp/minitest_reporters_report',
+        }
+      end
+
+      # Added to the top of the report file to be helpful.
+      #
+      # @return [String]
+      def report_header
+        "Samples: #{samples}\n\n"
+      end
+
+      # The report itself. Displays statistic about all runs, ideal for use with
+      # the Unix 'head' command. Listed in slowest average descending order.
+      #
+      # @return [String]
+      def report_body
+        previous_run.each_with_object([]) do |(description, timings), obj|
+          size = timings.size
+          sum  = timings.inject { |total, x| total + x }
+          avg  = (sum / size).round(9).to_s.ljust(12)
+          min  = timings.min.to_s.ljust(12)
+          max  = timings.max.to_s.ljust(12)
+
+          obj << "#{avg_label} #{avg} " \
+                 "#{min_label} #{min} " \
+                 "#{max_label} #{max} " \
+                 "#{des_label} #{description}\n"
+        end.sort.reverse.join
+      end
+
+      # @return [Hash]
+      def options
+        defaults.merge!(@options)
+      end
+
+      # @return [Hash<String => Array<Float>]
+      def previous_run
+        @previous_run ||= YAML.load_file(previous_runs_filename)
+      end
+
+      # @return [String] The path to the file which contains all the durations
+      #   for each test run. The previous runs file is in YAML format, using the
+      #   test name for the key and an array containing the time taken to run
+      #   this test for values.
+      #
+      # @return [String]
+      def previous_runs_filename
+        options[:previous_runs_filename]
+      end
+
+      # Returns a boolean indicating whether a previous runs file exists.
+      #
+      # @return [Boolean]
+      def previously_ran?
+        File.exist?(previous_runs_filename)
+      end
+
+      # @return [String] The path to the file which contains the parsed test
+      #   results. The results file contains a line for each test with the
+      #   average time of the test, the minimum time the test took to run,
+      #   the maximum time the test took to run and a description of the test
+      #   (which is the test name as emitted by Minitest).
+      def report_filename
+        options[:report_filename]
+      end
+
+      # A barbaric way to find out how many runs are in the previous runs file;
+      # this method takes the first test listed, and counts its samples
+      # trusting (naively) all runs to be the same number of samples. This will
+      # produce incorrect averages when new tests are added, so it is advised
```

Comment:
Hmm, I'm not sure I like the fact that it is user who is responsible for resetting the data.
Perhaps we could change this and either reset it for her or be able to report stats for both old and new tests.
WDYT?